### PR TITLE
Hosting: fix robots issues

### PR DESCRIPTION
**Pull request by Narretz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?**

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [ ] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [ ] Fix/Feature: Tests have been added; existing tests pass

**Other information**:

### CODE REVIEW

File: docs/config/processors/sitemap.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        id: 'sitemap.xml',
+        path: 'sitemap.xml',
+        outputPath: '../sitemap.xml',
+        template: 'sitemap.template.xml',
+        urls: docs.filter(function(doc) {
+          return doc.path &&
+            doc.outputPath &&
+            doc.outputPath.indexOf('index') !== 0 &&
+            doc.outputPath.indexOf('examples/') !== 0 &&
+            doc.outputPath.indexOf('ptore2e/') !== 0;
```
Comment (petebacondarwin) [inline docs/config/processors/sitemap.js]: You could do this with a regex:

```
!doc.outputPath.test(/^index|examples\/|ptore2e\//)
```

(@gkalpak would be proud of me :-D )

**Comment by IgorMinar** [comment]
lol

File: docs/config/processors/sitemap.js (hunk 1 above)
Comment (IgorMinar) [reply docs/config/processors/sitemap.js]: lol

**Review by IgorMinar — APPROVED** [review]
this looks reasonable to me, but let's wait @petebacondarwin approval as well

File: scripts/code.angularjs.org-firebase/public/robots.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 User-agent: *
 
-Disallow: /*docs/
+Disallow: /*docs*/
 Disallow: /*i18n/
 Disallow: /*.zip$
+Allow: /snapshot/docs/js/all-versions-data.js
```
Comment (petebacondarwin) [inline scripts/code.angularjs.org-firebase/public/robots.txt]: Apparently wildcards can make the resolution of ordering between rules undefined: https://developers.google.com/search/reference/robots_txt#order-of-precedence-for-group-member-records

Let's keep an eye on this after it goes live...

**Review by petebacondarwin — APPROVED** [review]

**Comment by IgorMinar** [comment]
actually, is `Allow: ` supported in robots.txt? I don't see any mention of it on http://www.robotstxt.org

I gave Pete access to the search console so that he can test the robots.txt there

**Comment by Narretz** [comment]
Google lists "allow": https://developers.google.com/search/reference/robots_txt?hl=en#allow

**Comment by petebacondarwin** [comment]
@IgorMinar 👍 

> actually, is Allow: supported in robots.txt? I don't see any mention of it on http://www.robotstxt.org

`robotstxt.org` is very out of date (~2007).

See

**Pull request merged**